src/JuliusSweetland.OptiKey/UI/Controls/Key.cs

```diff
@@ -47,35 +47,41 @@ private void OnLoaded(object sender, RoutedEventArgs routedEventArgs)
             var capturingStateManager = mainViewModel.CapturingStateManager;
 
             //Calculate KeyDownState
-            var keyStateSubscription = keyStateService.KeyDownStates[Value]
-                .OnPropertyChanges(kds => kds.Value)
-                .Subscribe(value => KeyDownState = value);
-            onUnloaded.Add(keyStateSubscription);
-            KeyDownState = keyStateService.KeyDownStates[Value].Value;
+            if (Value != null)
+            {
+                var keyStateSubscription = keyStateService.KeyDownStates[Value]
+                    .OnPropertyChanges(kds => kds.Value)
+                    .Subscribe(value => KeyDownState = value);
+                onUnloaded.Add(keyStateSubscription);
+            }
+            KeyDownState = (Value == null) ? KeyDownStates.Up : keyStateService.KeyDownStates[Value].Value;
 
             //Calculate SelectionProgress and SelectionInProgress
-            var keySelectionProgressSubscription = keyStateService.KeySelectionProgress[Value]
-                .OnPropertyChanges(ksp => ksp.Value)
-                .Subscribe(value =>
-                {
-                    SelectionProgress = value;
-                    SelectionInProgress = value > 0d;
-                });
-            onUnloaded.Add(keySelectionProgressSubscription);
-            var progress = keyStateService.KeySelectionProgress[Value].Value;
+            if (Value != null)
+            {
+                var keySelectionProgressSubscription = keyStateService.KeySelectionProgress[Value]
+                    .OnPropertyChanges(ksp => ksp.Value)
+                    .Subscribe(value =>
+                    {
+                        SelectionProgress = value;
+                        SelectionInProgress = value > 0d;
+                    });
+                onUnloaded.Add(keySelectionProgressSubscription);
+            }
+            var progress = (Value == null) ? 0 : keyStateService.KeySelectionProgress[Value].Value;
             SelectionProgress = progress;
             SelectionInProgress = progress > 0d;
 
             //Calculate IsEnabled
-            Action calculateIsEnabled = () => IsEnabled = keyStateService.KeyEnabledStates[Value];
+            Action calculateIsEnabled = () => IsEnabled = Value != null && keyStateService.KeyEnabledStates[Value];
```

Comment:
@kirstymcnaught this alteration is causing the blank CK keys to be black instead of coloured. Changing this line to 
```
Action calculateIsEnabled = () => IsEnabled = (Value != null || HasImage) && keyStateService.KeyEnabledStates[Value];
```
fixes this and I believe it should not effect anything else.

Reply:
@kirstymcnaught I made a PR (https://github.com/kirstymcnaught/OptiKeyMinecraft/pull/1) with this addition along with a modification to the debug logging for the CK Back key. I'm not completely sure if that is the correct procedure so if it's not please ignore it.

Reply:
Thanks! I've answered below.